### ISSUE TITLE
Banding artifacts 

### ISSUE THREAD

**Issue by Ilya87** [OpenVisualCloud/SVT-HEVC#158]
Os - Antergos x86_64? SVT-HEVC latest from git
https://yadi.sk/i/qBkjd3TzJ_yBQQ - original h.264 file from blu-ray (was converted to YUV)
https://yadi.sk/d/K6LBH3HdDVF6zQ - encoded version

Encoding options are:
`-w 1920 -h 1080 -tune 1 -intra-period 239 -rc 0 -profile 2 -encMode 3 -fps-num 24 -fps-denom 1 -encMode 3 -lad 49 -q 15 -n 751 -b out.hevc`

Example screenshots (artifacts on faces, eyes):
![vlcsnap-2019-03-25-1](https://user-images.githubusercontent.com/1446573/54888837-e7675e80-4eb1-11e9-975f-93737584b4a1.png)
![vlcsnap-2019-03-25-2](https://user-images.githubusercontent.com/1446573/54888838-e7fff500-4eb1-11e9-9be4-0bac515e9693.png)
![vlcsnap-2019-03-25-3](https://user-images.githubusercontent.com/1446573/54888839-e7fff500-4eb1-11e9-80f1-5f97f60c2a41.png)

**Comment by 1480c1** [comment]
Good taste. However, the ss are so dark, that it's so hard to see the banding. Also, why the ? after Antergos x86_64?

**Comment by Ilya87** [comment]
@1480c1 ? is a typo (corrected in post), comma should be used. And not, when you'll see screenshots fullscreen using 24" monitor they are clearly seen, moreover this screenshots from still scene, moreover x264 doesn't produce such artifacts.

**Comment by 1480c1** [comment]
That is true, I saw it when I zoomed in.

**Comment by tianjunwork** [comment]
@Ilya87 yes, I can see the artifacts on the face. 
6.1MBx8/(751/24) = about 1.6Mbps which is not very high for 1080p. What is the bitrate of x264 bitstream?

**Comment by Ilya87** [comment]
@tianjunwork, yes, I'm sorry - at this bitrate x264 also provides comparable banding artifacts, BUT at comparable bitrate x264 is better
x264 veryslow_qp15_bitrate_3549
https://yadi.sk/i/uBj5eDI_m7-Pcw
SVT-HEVC -q12 target bitrate is 3108
https://yadi.sk/i/gFZWSRU0kXDZPg
Banding artifacts are still the problem.

**Comment by Ilya87** [comment]
![svt_q12_11 3](https://user-images.githubusercontent.com/1446573/54949117-68c0fe80-4f4f-11e9-9e2a-8593afbe268f.png)
![svt_q12_24 5](https://user-images.githubusercontent.com/1446573/54949118-68c0fe80-4f4f-11e9-882c-32c4b15a995a.png)
![svt_q12_28 5](https://user-images.githubusercontent.com/1446573/54949120-68c0fe80-4f4f-11e9-909d-f9b8bce7e75b.png)

**Comment by Ilya87** [comment]
Compare with x264 veryslow -crf 16.6 (bitrate is 3088)
https://yadi.sk/i/2Ow-PxJZdqcmUg
Comparable encoding speed, same bitrate, banding artifacts are much less.

**Comment by tianjunwork** [comment]
What is the tune mode used by x264? Could you try -tune 0/2 with SVT?

**Comment by Ilya87** [comment]
@tianjunwork, for x264 I used ffmpeg with -c:v libx264 -preset veryslow -qp 15. No additional options. 
For crf mode -c:v libx264 -preset veryslow  -crf 16.6

**Comment by tianjunwork** [comment]
I tried with different tune mode(all other configs are the same), tune0 is slightly better for banding artifacts. I took the snapshot from Intel media analyzer tool. All of the tune modes are not as obvious as viewed in VLC.
tune0
![tune0_m3](https://user-images.githubusercontent.com/4551215/55031149-2ccb7f00-4fcb-11e9-91cf-1fef5b0e1cc8.JPG)
tune1
![tune1_m3](https://user-images.githubusercontent.com/4551215/55031150-2ccb7f00-4fcb-11e9-9740-3a764716ee98.JPG)
tune2
![tune2_m3](https://user-images.githubusercontent.com/4551215/55031151-2d641580-4fcb-11e9-90eb-32ddfa944aa0.JPG)

**Comment by Ilya87** [comment]
@tianjunwork, such clearly seen artifacts as I checked have divx265, x265, turing, SVT-HEVC encoders (there is open issue about banding in x265 bug tracker).
While x264 has much less banding artifacts.
So, my conclusion is that all HEVC standard encoders can't encode without such artifacts by design.
If you wish, I can upload encoded materials (same sample as here) with settings description, VMAF values, bitrates.

**Comment by tianjunwork** [comment]
Hevc introduced SAO filter applied after the standard deblocking filter; it’s designed to increase picture quality and reduce banding and ringing artifacts at some cost of encoding speed. Generally speaking, hevc should perform better than avc.
I think SVT has room to further reduce banding artifacts. Please do upload your material:) which will help us debugging later. Thanks!
I will mark this issue as a low priority task based on our current resources.

**Comment by Ilya87** [comment]
https://yadi.sk/d/KizxwX0-Tu-WRg - encoded files with results in ods

**Comment by tianjunwork** [comment]
@Ilya87 Thx for the files.
You can also turn off SAO for SVT. For all the encoding tools, trade off between video quality and speed has to be made. With SAO turned on, in order not to affect speed so much, parameters of SAO is tuned for each every encoder. So you observe the difference among each encoder output.

**Comment by Ilya87** [comment]
@tianjunwork, the problem is not in SAO turned on or off (that doesn't affect much), but in the same banding artifacts of all HEVC encoders for high quality encodes.
